src/components/search-by-demographics/search-by-demographics.test.tsx

```diff
@@ -0,0 +1,85 @@
+import React from "react";
+
+import { render, cleanup, fireEvent } from "@testing-library/react";
+import dayjs from "dayjs";
+
+import { SearchByDemographics } from "./search-by-demographics.component";
+
+const mockQuery = {
+  query: {
+    type: "org.openmrs.module.reporting.dataset.definition.PatientDataSetDefinition",
+    columns: [
+      {
+        name: "firstname",
+        key: "reporting.library.patientDataDefinition.builtIn.preferredName.givenName",
+        type: "org.openmrs.module.reporting.data.patient.definition.PatientDataDefinition",
+      },
+      {
+        name: "lastname",
+        key: "reporting.library.patientDataDefinition.builtIn.preferredName.familyName",
+        type: "org.openmrs.module.reporting.data.patient.definition.PatientDataDefinition",
+      },
+      {
+        name: "gender",
+        key: "reporting.library.patientDataDefinition.builtIn.gender",
+        type: "org.openmrs.module.reporting.data.patient.definition.PatientDataDefinition",
+      },
+      {
+        name: "age",
+        key: "reporting.library.patientDataDefinition.builtIn.ageOnDate.fullYears",
+        type: "org.openmrs.module.reporting.data.patient.definition.PatientDataDefinition",
+      },
+      {
+        name: "patientId",
+        key: "reporting.library.patientDataDefinition.builtIn.patientId",
+        type: "org.openmrs.module.reporting.data.patient.definition.PatientDataDefinition",
+      },
+    ],
+    rowFilters: [
+      {
+        key: "reporting.library.cohortDefinition.builtIn.males",
+        type: "org.openmrs.module.reporting.dataset.definition.PatientDataSetDefinition",
+      },
+      {
+        key: "reporting.library.cohortDefinition.builtIn.ageRangeOnDate",
+        parameterValues: {
+          minAge: "10",
+          maxAge: "20",
+        },
+        type: "org.openmrs.module.reporting.dataset.definition.PatientDataSetDefinition",
+      },
+      {
+        key: "reporting.library.cohortDefinition.builtIn.diedDuringPeriod",
+        parameterValues: {
+          endDate: "2022-07-09T17:12:47+05:30",
+        },
+        type: "org.openmrs.module.reporting.dataset.definition.PatientDataSetDefinition",
+      },
+    ],
+    customRowFilterCombination: "1 AND 2 AND NOT 3",
+  },
+};
+
+describe("Test the search by demographics component", () => {
+  afterEach(cleanup);
+
+  it("should be able to select input values", () => {
+    const setSearchParams = jest.fn();
+    const setQueryDescription = jest.fn();
+    const { getByTestId } = render(
+      <SearchByDemographics
+        setQueryDescription={setQueryDescription}
+        setSearchParams={setSearchParams}
```

Comment:
Didn't notice this earlier. These two props' names look ambiguous to me. 

shall we rename it with a proper convention? ex: `onX`